src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/PbfToRawAtlasCommand.java

```diff
@@ -0,0 +1,177 @@
+package org.openstreetmap.atlas.utilities.command.subcommands;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas.AtlasSerializationFormat;
+import org.openstreetmap.atlas.geography.atlas.raw.creation.RawAtlasGenerator;
+import org.openstreetmap.atlas.geography.sharding.CountryShard;
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.streaming.resource.FileSuffix;
+import org.openstreetmap.atlas.utilities.command.AtlasShellToolsException;
+import org.openstreetmap.atlas.utilities.command.abstractcommand.CommandOutputDelegate;
+import org.openstreetmap.atlas.utilities.command.abstractcommand.OptionAndArgumentDelegate;
+import org.openstreetmap.atlas.utilities.command.parsing.ArgumentArity;
+import org.openstreetmap.atlas.utilities.command.parsing.ArgumentOptionality;
+import org.openstreetmap.atlas.utilities.command.parsing.OptionOptionality;
+import org.openstreetmap.atlas.utilities.command.subcommands.templates.MultipleOutputCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author samgass
+ */
+public class PbfToRawAtlasCommand extends MultipleOutputCommand
+{
+    private static final Logger logger = LoggerFactory.getLogger(PbfToRawAtlasCommand.class);
+
+    // The filetype for PBFs-- will strip this from the filename to make the Atlas filename
+    private static final String PBF_FILETYPE = ".pbf";
+
+    // The hint for the input path for the PBF file(s) to convert
+    private static final String PBF_PATH_HINT = "pbf";
+
+    // The country name for the country shards Atlas file(s) to output
+    private static final String COUNTRY_NAME = "countryName";
+
+    private static final String COUNTRY_NAME_DESCRIPTION = "The country for the shard to build";
+
+    private final OptionAndArgumentDelegate optionAndArgumentDelegate;
+    private final CommandOutputDelegate outputDelegate;
+
+    private List<File> pbfs;
+
+    public static void main(final String[] args)
+    {
+        new PbfToRawAtlasCommand().runSubcommandAndExit(args);
+    }
+
+    public PbfToRawAtlasCommand()
+    {
+        super();
+        this.optionAndArgumentDelegate = this.getOptionAndArgumentDelegate();
+        this.outputDelegate = this.getCommandOutputDelegate();
+    }
+
+    @Override
+    public int execute()
+    {
+        // set up the output path from the parent class
+        final int code = super.execute();
```

Comment:
Another little wart in my design....

What I really want instead of having abstract super classes is to have some way of defining command templates. Since you can only inherit from one superclass, having the command templates (like `MultipleOutputCommand`) be abstract classes is limiting. And it leads to design warts like the one here.

Reply:
I'm on board with this actually. realistically, the parent class here legitimately is going to be executing code beforehand, combined with the extra behavior implemented by the subclass-- that seems to fit the abstract class model nicely.

Reply:
I suppose. I just worry that down the line we may desire more flexibility - what if we want a `MultipleOutputCommand` that is also a `DoohickyCommand`. But I suppose for now this design keeps things simple.